src/components/EventForm/EventForm.tsx

```diff
@@ -0,0 +1,186 @@
+    format(rest.startTime || Date.now(), 'HH:mm')
+  );
+  const [endTime, setEndTime] = useState(
+    format(rest.endTime || Date.now(), 'HH:mm')
+  );
+  const [description, setDescription] = useState(rest.description || '');
+  const [eventType, setEventType] = useState(rest.eventType || '');
+  const [location, setLocation] = useState(rest.location || '');
+
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [hasSubmitted, setHasSubmitted] = useState(false);
+  const [error, setError] = useState<string | null>(null);
+
+  const isFormValid = name && description && eventType;
+  const isSubmitDisabled = !isFormValid || isSubmitting;
+
+  const handleSubmit = useCallback(async () => {
+    setIsSubmitting(true);
+    setError(null);
+
+    try {
+      await onSubmit({
+        id: rest.id,
+        name,
+        description,
+        startTime: new Date(date + ' ' + startTime),
+        endTime: new Date(date + ' ' + endTime),
+        imageUrl,
+        imageAlt,
+        location,
+        eventType,
+      });
+    } catch (error) {
+      setError(error.message);
+    } finally {
+      setIsSubmitting(false);
+    }
+    setHasSubmitted(true);
+  }, [
+    name,
+    description,
+    date,
+    startTime,
+    endTime,
+    imageUrl,
+    imageAlt,
+    location,
+    eventType,
+    setIsSubmitting,
+    setError,
+    onSubmit,
+    rest,
+  ]);
+
+  return (
+    <EventFormWrapper>
+      <h1>{rest.headerTitle}</h1>
+      <Form>
+        <TextField
+          name="name"
+          label="Event name"
+          value={name}
+          onChange={setName}
+        />
+        <DateField
+          name="date"
+          label="Date"
+          value={date}
+          onChange={(date: string) => setDate(date)}
+        />
+        <TimeField
+          name="time"
+          label="Start time"
+          value={startTime}
+          onChange={setStartTime}
+        />
+        <TimeField
+          name="time"
+          label="End time"
+          value={endTime}
+          onChange={setEndTime}
+        />
+        <TextAreaField
+          name="description"
+          label="Description"
+          value={description}
+          onChange={setDescription}
+        />
+        <TextField
+          name="location"
+          label="Location"
+          value={location}
+          onChange={setLocation}
+        />
+        <TextField
+          name="image"
+          label="Image link"
+          value={imageUrl}
+          onChange={setImageUrl}
+        />
+        <TextField
+          name="imageAlt"
+          label="Image text"
+          value={imageAlt}
+          onChange={setImageAlt}
+        />
+        <SelectField
+          name="eventType"
+          label="Event type"
+          value={eventType}
+          onChange={setEventType}
+          options={EVENT_OPTIONS}
+        />
+        <div>
+          <Button disabled={isSubmitDisabled} onClick={handleSubmit}>
+            Create
```

Comment:
Hmm, we'll need this to be `Update` instead of `Create` when a user is updating an existing event 🤔. Will probably need to pass down an extra prop stating whether the event is being created or updated.